stdlib/public/Concurrency/AsyncStream.swift

```diff
@@ -141,22 +215,46 @@ public struct AsyncStream<Element> {
 
   let produce: () async -> Element?
 
-  /// Construct a AsyncStream buffering given an Element type.
-  ///
-  /// - Parameter elementType: The type the AsyncStream will produce.
-  /// - Parameter maxBufferedElements: The maximum number of elements to
-  ///   hold in the buffer past any checks for continuations being resumed.
-  /// - Parameter build: The work associated with yielding values to the 
-  ///   AsyncStream.
-  ///
-  /// The maximum number of pending elements limited by dropping the oldest
-  /// value when a new value comes in if the buffer would excede the limit
-  /// placed upon it. By default this limit is unlimited.
-  ///
-  /// The build closure passes in a Continuation which can be used in
-  /// concurrent contexts. It is thread safe to send and finish; all calls are
-  /// to the continuation are serialized, however calling this from multiple
-  /// concurrent contexts could result in out of order delivery.
+  /// Constructs an asynchronous stream for an element type, using the
+  /// specified buffering policy and element-producing closure.
+  ///
+  /// - Parameter elementType: The type of element the `AsyncStream`
+  ///   produces
+  /// - Parameter limit: The maximum number of elements to hold in the buffer.
+  ///   By default, this value is unlimited. Use a
+  ///   `Continuation.BufferingPolicy` to buffer a specified number of oldest
+  ///   or newest elements.
+  /// - Parameter build: A custom closure that yields values to the
+  ///   `AsyncStream`. This closure receives a `AsyncStream.Continuation`
+  ///   instance that it uses to provide elements to the stream and terminate the
+  ///   stream when finished.
+  ///
+  /// The `AsyncStream.Contuation` received by the `build` closure is appopriate
+  /// for use in concurrent contexts. It is thread safe to send and finish; all
+  /// calls are to the continuation are serialized, however calling this from
+  /// multiple concurrent contexts could result in out of order delivery.
+  ///
+  /// The following example shows an `AsyncStream` created with this
+  /// initializer that produces random numbers on a one-second interval. When
+  /// a private `keepRunning` variable becomes `false`, the inner `while` loop
+  /// exits and the stream finishes.
+  ///
+  ///     let stream = AsyncStream<Int>(
+  ///         Int.self, bufferingPolicy: .bufferingNewest(5)) { continuation in
+  ///             Task.detached {
+  ///                 while (keepRunning) {
```

Comment:
```suggestion
  ///                 while keepRunning {
```

C requires parens here; Swift doesn't.